fishtank/src/backend/docker.ts

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ChildProcess, execFile, ExecFileOptions } from 'child_process'
+import { promisify } from 'util'
+
+const DEFAULT_COMMAND = 'docker'
+
+type ExecFilePromiseReturn = { child: ChildProcess; stdout: string; stderr: string }
+
+type ExecFilePromiseError = {
+  cmd: string
+  code: number | string
+  stdout: string
+  stderr: string
+}
+
+const execChildProcess = promisify<
+  string,
+  readonly string[],
+  ExecFileOptions,
+  ExecFilePromiseReturn
+>(execFile)
+
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access */
+const isExecFilePromiseError = (e: any): e is ExecFilePromiseError => {
+  return (
+    !!e &&
+    typeof e === 'object' &&
+    typeof e['cmd'] === 'string' &&
+    (typeof e['code'] === 'number' || typeof e['code'] === 'string') &&
+    typeof e['stdout'] === 'string' &&
+    typeof e['stderr'] === 'string'
+  )
+}
+
+export class DockerError extends Error {
+  constructor(
+    public readonly cmd: string,
```

Comment:
Can you rename it to `command` so it keep consistent with Docker `command` variable and constructor. Currently in this file there are both `command` and `cmd` used and they are referring to the same string value, it would be nice to unify the name

Reply:
It'd be better to rename rename `command` to `executable`. The command in `DockerError` is not the same command held by the `Docker` class.

Reply:
Sounds good